### ISSUE TITLE
Allow configuration for LocalDomain

### ISSUE THREAD

**Issue by afoone** [swiftmailer/swiftmailer#453]
The local domain is 'guessed' in the method:

```
/** Try to determine the hostname of the server this is run on */
private function _lookupHostname()
{
    if (!empty($_SERVER['SERVER_NAME'])
        && $this->_isFqdn($_SERVER['SERVER_NAME']))
    {
        $this->_domain = $_SERVER['SERVER_NAME'];
    } elseif (!empty($_SERVER['SERVER_ADDR'])) {
        $this->_domain = sprintf('[%s]', $_SERVER['SERVER_ADDR']);
    }
}
```

There's a problem with this: in Outlook.com , ipv6 is not supported for HELO command, then I need to set 'setLocalDomain' to 127.0.0.1 each time. Can you add a configuration option to avoid this issue?

**Comment by xelan** [comment]
The AbstractSmtpTransport has a method setLocalDomain(), which sets the same property as _lookupHostname().

**Comment by ThYpHo0n** [comment]
+1 this is really annoying, e.g. if you use regex in nginx server_name configuration.

**Comment by kanavis** [comment]
![image](https://cloud.githubusercontent.com/assets/5455322/21924099/99be02a6-d988-11e6-80ed-bb55546d5e36.png)
No there is a point (and it's written above)

**Comment by c960657** [comment]
Fixed in #672 and/or #799.

**Comment by kanavis** [comment]
Great! Many thanks!
